app/controllers/documents_controller.rb

```diff
@@ -122,7 +122,7 @@ def document_id
   end
 
   def ensure_document_edits_unlocked
-    render plain: "forbidden", status: :forbidden and return unless @planning_application.can_validate?
+    render plain: "forbidden", status: :forbidden and return unless @planning_application.can_validate? || @planning_application.publish_complete?
```

Comment:
What about a new method in PlanningApplication that represents this state, since we use it twice?

It might be clearer that way — especially as `publish_complete?` is an alias for `determined?`, but `can_validate?` includes `!closed_or_cancelled?` and `closed_or_cancelled?` includes `determined?`. This is all correct but might get confusing because of the number of different states we have.

Reply:
I wanted to use another method, but when I tried tweaking the existing one or adding a new one, something did not work as it should, as we have many `ifs`.

Reply:
What I mean is
 
```suggestion
    render plain: "forbidden", status: :forbidden and return unless @planning_application.can_edit_documents?
```

And then on PlanningApplication:

```ruby
def can_edit_documents?
  can_validate? || publish_complete?
end
```